my_site/blog.php

```diff
@@ -0,0 +1,160 @@
+<?php
+// start the session if it hasn’t been initialized yet
+if (session_status() === PHP_SESSION_NONE) {
+    session_start();
+    var_dump($_SESSION);
+
+}
+
+// header and the blog data functions
+include 'header.php'; 
+require_once 'blog_model.php'; 
+
+// taking all posts from the JSON file
+$posts = load_all_posts();
+
+// checking if the user is logged in ro not
+// Allow admin view via session OR a query flag (?admin=1) for environments without a dedicated blog login.
+// This does NOT bypass your security in the controller; it only reveals UI controls. 
+$is_admin_session = isset($_SESSION['is_logged_in']) && $_SESSION['is_logged_in'] === true;
+$is_admin_query   = (isset($_GET['admin']) && $_GET['admin'] === '1');
+$is_admin = $is_admin_session || $is_admin_query;
+?>
+
+<div class="blog-hero" style="text-align:center; padding:40px 20px; background:#b2d8d8; border-radius:8px; margin-bottom:20px;">
+    <h1 style="font-size:2.5rem; margin:0; color:#333; font-family:'Georgia',serif;">The Developer's Blog</h1>
+    <p style="font-size:1.2rem; color:#555; margin-top:10px;">Documenting the journey of building a robust Web Architecture.</p>
+</div>
+
+<!-- Theme toggle button -->
+<div style="text-align:center; margin:20px;">
+    <button id="theme-toggle" 
+            style="padding:10px 20px; border:none; border-radius:5px; cursor:pointer; background:#333; color:#fff;">
+        Toggle Light/Dark Theme
+    </button>
+</div>
+
+<?php if ($is_admin): ?>
+<!-- to show the add post button but only for logged-in users -->
+<div style="text-align: center; margin-top: 20px; margin-bottom: 20px;">
+    <a href="blog_add.php" 
+       style="background-color: #007bff; color: white; padding: 10px 20px; text-decoration: none; border-radius: 4px; display: inline-block;">
+        + Add New Post
+    </a>
+</div>
+<?php endif; ?>
+
+<div class="blog-container">
+
+    <main class="blog-main">
+        <?php
+        if (!empty($posts)) {
+            // switching the order so that the last posts appears first
+            // If your JSON is an associative array keyed by ID, array_values ensures proper reindexing for array_reverse.
```

Comment:
Same comment here about AI. This is the reason why I was asking your to add many commits when you work. To make sure that the work was improved by you, step by step, not simply copy-pasted from AI. I can't know what portion of this work was created by you, and I only see a single commit: "Done".